cr.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  Netguru iOS code review task
+//
+
+class paymentViewController: UIViewController {
+    var delegate: PaymentViewControllerDelegate
+    let customView = PaymentView()
+    let payment: Payment?
+
+    func callbacks() {
+        view.backgroundColor = UIColor(displayP3Red: 1, green: 1, blue: 1, alpha: 1)
+        customView.didTapButton = {
+            if let payment = self.payment {
+                self.delegate.didFinishFlow(amount: payment.amount
+                                            currency: payment.currency)
+            }
+        }
+    }
```

Comment:
- Maybe better to declare as **_private_**
- You are calling **_setupCallbacks_** later, so, please rename it
- also please move in some other part after lifecycle functions and inits
- It will be cleaner to use Constants and declared colours that can be reused, look into [swiftGen](https://github.com/SwiftGen/SwiftGen) to help you with
- avoid retain cycles when using closures, use **[unowned self]**  -> this closures will be never called when self already deinited so you can avoid using weak and optional **self**
- when calling delegate you are missing coma after **_payment.amount_**
- as delegate should be weak and optional here, then you need to use optional chaining

Below full code snippet how this closure should look like:
```suggestion
    private func setupCallbacks() {
        view.backgroundColor = UIColor(displayP3Red: 1, green: 1, blue: 1, alpha: 1)
        customView.didTapButton = { [unowned self] in
            if let payment = self.payment {
                self.delegate?.didFinishFlow(amount: payment.amount,
                                            currency: payment.currency)
            }
        }
    }
```